packages/contracts/src/EmailSigner.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import {EmailProof} from "./utils/Verifier.sol";
+import {IDKIMRegistry} from "@zk-email/contracts/DKIMRegistry.sol";
+import {IVerifier, EmailProof} from "./interfaces/IVerifier.sol";
+import {CommandUtils} from "./libraries/CommandUtils.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {IEmailAuth, EmailAuthMsg} from "./interfaces/IEmailAuth.sol";
+import {IERC1271} from "./interfaces/IERC1271.sol";
+import {ERC1271} from "./libraries/ERC1271.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/// @title Email Authentication/Authorization Contract for Signature-like Usage
+/// @notice This contract provides a signature-like authentication mechanism using emails.
+/// Similar to how ECDSA signatures work, this contract only verifies the authenticity
+/// of an email command without handling replay protection or nullifiers - these should
+/// be implemented at the application level.
+/// @dev Unlike EmailAuth.sol which handles nullifiers internally, this contract is designed
+/// to be used like a signature verification mechanism where the calling contract manages
+/// its own replay protection.
+contract EmailSigner is IEmailAuth, Initializable {
+    /// Unique identifier for owner of this contract defined as a hash of an email address and an account code.
+    bytes32 public accountSalt;
+    /// An instance of the DKIM registry contract.
+    address public dkimRegistryAddr;
+    /// An instance of the Verifier contract.
+    address public verifierAddr;
+    /// The templateId of the sign hash command.
+    uint256 public templateId;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the contract with an initial owner, account salt, DKIM registry address, and verifier address.
+    /// @param _accountSalt The account salt to derive CREATE2 address of this contract.
+    /// @param _dkimRegistryAddr The address of the DKIM registry contract.
+    /// @param _verifierAddr The address of the verifier contract.
+    /// @param _templateId The templateId of the sign hash command.
+    function initialize(
+        bytes32 _accountSalt,
+        address _dkimRegistryAddr,
+        address _verifierAddr,
+        uint256 _templateId
+    ) public initializer {
+        accountSalt = _accountSalt;
+        templateId = _templateId;
+
+        if (_dkimRegistryAddr == address(0))
```

Comment:
Couldn't see a unit test for InvalidDKIMRegistryAddress & InvalidVerifierAddress